cranelift/filetests/src/test_zkasm.rs

```diff
@@ -0,0 +1,187 @@
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use cranelift_codegen::entity::EntityRef;
+    use cranelift_codegen::ir::function::FunctionParameters;
+    use cranelift_codegen::ir::ExternalName;
+    use cranelift_codegen::isa::zkasm;
+    use cranelift_codegen::{settings, FinalizedMachReloc, FinalizedRelocTarget};
+    use cranelift_wasm::{translate_module, DummyEnvironment};
+
+    fn generate_preamble(start_func_index: usize) -> Vec<String> {
+        let mut program: Vec<String> = Vec::new();
+        program.push("start:".to_string());
+        program.push("  zkPC + 2 => RR".to_string());
+        program.push(format!("  :JMP(function_{})", start_func_index));
+        program.push("  :JMP(finalizeExecution)".to_string());
+        program
+    }
+
+    fn generate_postamble() -> Vec<String> {
+        let mut program: Vec<String> = Vec::new();
+        program.push("finalizeExecution:".to_string());
+        program.push("  ${beforeLast()}  :JMPN(finalizeExecution):".to_string());
+        program.push("                   :JMP(start)".to_string());
+        program
+    }
+
+    fn fix_relocs(
```

Comment:
Could you also slap a `FIXME` on this? This test runner doesn't seem like a great place for relocation fixup and other such work. I am also suspicious of our need to have relocations at all!

Reply:
I agree, added a TODO.